Add indentation rule for closing tag of multi-line jsx

rsolomon opened the pull request:
For issue #1206:

This rule validates that the closing tag for a multi-line jsx component matches the indentation of the opening tag.

ljharb commented on tests/lib/rules/jsx-closing-tag-location.js:
```diff
@@ -0,0 +1,56 @@
+const rule = require('../../../lib/rules/jsx-closing-tag-location');
+const {RuleTester} = require('eslint');
+const parserOptions = {
+  ecmaVersion: 8,
```
These tests don't need an ecmaVersion specified, nor rest/spread enabled

rsolomon replied:
Oops! Will fix.

rsolomon commented:
@ljharb I think I've addressed your concerns. The one situation this doesn't fix is the following:

```
<App
  foo="bar">baz</App>
```

It will fix it to:

```
<App
  foo="bar">baz
</App>
```

When ideally it would be fixed to:

```
<App
  foo="bar">
  baz
</App>
```

I opted not to handle it in this rule because I imagine multiline children location should be handled in a separate rule entirely (maybe it already is?)

ljharb commented:
I agree with you that it should be handled in a separate rule.

rsolomon commented:
Pardon my ignorance (this is the first eslint rule I've written), but isn't that what the `output` attributes of the `invalid` tests are for?

HHogg commented:
@rsolomon Would this rule also cover the situation below? 

```jsx
// Considered a warning 
<App>
  { condition && <Component>
    ...
  </Component> }
</App>

// Not considered a warning 
<App>
  { condition && (
    <Component>
      ...
    </Component>    
  ) }
</App>
```
